libmisc/utmp.c

```diff
@@ -215,8 +214,7 @@ static void updwtmpx (const char *filename, const struct utmpx *utx)
 	} else if (   (NULL != ut)
 	           && ('\0' != ut->ut_host[0])) {
 		hostname = (char *) xmalloc (sizeof (ut->ut_host) + 1);
-		strncpy (hostname, ut->ut_host, sizeof (ut->ut_host));
-		hostname[sizeof (ut->ut_host)] = '\0';
+		shdw_strscpy (hostname, ut->ut_host, sizeof (ut->ut_host) + 1);
```

Comment:
xstrdup()?

Reply:
BTW, just to confirm that there are no other places than the 2 you suggested, I run the following:

```c
$ grep -rnI -A5 xmalloc | grep -i -B5 str.cpy
libmisc/utmp.c-212-		strcpy (hostname, host);
libmisc/utmp.c-213-#ifdef HAVE_STRUCT_UTMP_UT_HOST
libmisc/utmp.c-214-	} else if (   (NULL != ut)
libmisc/utmp.c-215-	           && ('\0' != ut->ut_host[0])) {
libmisc/utmp.c:216:		hostname = (char *) xmalloc (sizeof (ut->ut_host) + 1);
libmisc/utmp.c-217-		shdw_strscpy (hostname, ut->ut_host, sizeof (ut->ut_host) + 1);
--
libmisc/utmp.c-357-#ifdef HAVE_STRUCT_UTMP_UT_HOST
libmisc/utmp.c-358-	} else if (   (NULL != ut)
libmisc/utmp.c-359-	           && (NULL != ut->ut_host)
libmisc/utmp.c-360-	           && ('\0' != ut->ut_host[0])) {
libmisc/utmp.c:361:		hostname = (char *) xmalloc (sizeof (ut->ut_host) + 1);
libmisc/utmp.c-362-		shdw_strscpy (hostname, ut->ut_host, sizeof (ut->ut_host) + 1);
```

Reply:
Oh, wait, there are a few more, when we consider `strcpy(3)`:

```c
$ grep -rnI -A5 x.alloc | grep -i -B5 -e str.cpy -e strcpy
src/su.c-245-		targs[n_args+2] = NULL;
src/su.c-246-		while (1 != n_args) {
--
src/su.c:1180:		arg0 = xmalloc (strlen (cp) + 2);
src/su.c-1181-		arg0[0] = '-';
src/su.c-1182-		strcpy (arg0 + 1, cp);
--
src/login.c-934-			if ('\0' == username[0]) {
src/login.c-935-				/* Prompt for a new login */
--
src/passwd.c:520:		char *newpw = xmalloc (strlen (cp) + 2);
src/passwd.c-521-
src/passwd.c-522-		strcpy (newpw, "!");
--
libmisc/utmp.c-133-
libmisc/utmp.c-134-	endutxent ();
libmisc/utmp.c-135-
--
libmisc/utmp.c:211:		hostname = (char *) xmalloc (strlen (host) + 1);
libmisc/utmp.c-212-		strcpy (hostname, host);
libmisc/utmp.c-213-#ifdef HAVE_STRUCT_UTMP_UT_HOST
libmisc/utmp.c-214-	} else if (   (NULL != ut)
libmisc/utmp.c-215-	           && ('\0' != ut->ut_host[0])) {
libmisc/utmp.c:216:		hostname = (char *) xmalloc (sizeof (ut->ut_host) + 1);
libmisc/utmp.c-217-		shdw_strscpy (hostname, ut->ut_host, sizeof (ut->ut_host) + 1);
--
libmisc/utmp.c-229-
libmisc/utmp.c-230-
libmisc/utmp.c-231-#ifdef HAVE_STRUCT_UTMP_UT_TYPE
--
libmisc/utmp.c:355:		hostname = (char *) xmalloc (strlen (host) + 1);
libmisc/utmp.c-356-		strcpy (hostname, host);
libmisc/utmp.c-357-#ifdef HAVE_STRUCT_UTMP_UT_HOST
libmisc/utmp.c-358-	} else if (   (NULL != ut)
libmisc/utmp.c-359-	           && (NULL != ut->ut_host)
libmisc/utmp.c-360-	           && ('\0' != ut->ut_host[0])) {
libmisc/utmp.c:361:		hostname = (char *) xmalloc (sizeof (ut->ut_host) + 1);
libmisc/utmp.c-362-		shdw_strscpy (hostname, ut->ut_host, sizeof (ut->ut_host) + 1);
--
libmisc/xmalloc.c-31-	void *ptr;
libmisc/xmalloc.c-32-
libmisc/xmalloc.c-33-	ptr = malloc (size);
libmisc/xmalloc.c-34-	if (NULL == ptr) {
--
libmisc/xmalloc.c:45:	return strcpy (xmalloc (strlen (str) + 1), str);
--
libmisc/xmalloc.c-48-void xfree(void *ap)
libmisc/xmalloc.c-49-{
libmisc/xmalloc.c-50-	if (ap) {
--
libmisc/obscure.c:161:	wrapped = xmalloc (strlen (oldmono) * 2 + 1);
libmisc/obscure.c-162-	strcpy (wrapped, oldmono);
```